apps/voting/contracts/Voting.sol

```diff
@@ -164,6 +179,36 @@ contract Voting is IForwarder, AragonApp {
         _vote(_voteId, _supports, msg.sender, _executesIfDecided);
     }
 
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of `_principal`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @param _principal Address of the principal voting on behalf of
+    * @param _voteId Id for vote
+    * @param _supports Whether the representative supports the vote
+    */
+    function voteOnBehalfOf(address _principal, uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        require(_canVoteOnBehalfOf(_voteId, _principal, msg.sender), ERROR_REPRESENTATIVE_CANT_VOTE);
+        _vote(_voteId, _supports, _principal, false);
+    }
+
+    /**
+    * @notice Vote on behalf of many principals
+    * @param _principals Addresses of the principals voting on behalf of
+    * @param _voteIds Ids for each of the proxied votes
+    * @param _supports Whether the representative supports each of the proxied votes
+    */
+    function voteOnBehalfOfMany(address[] _principals, uint256[] _voteIds, bool[] _supports) external isInitialized {
+        require(_principals.length == _voteIds.length && _voteIds.length == _supports.length, ERROR_INVALID_DELEGATES_INPUT_LEN);
+        require(_principals.length <= MAX_VOTES_DELEGATION_SET_LENGTH, ERROR_DELEGATES_EXCEEDS_MAX_LEN);
+
+        for (uint256 i = 0; i < _principals.length; i++) {
+            if (_voteExists(_voteIds[i]) && _canVoteOnBehalfOf(_voteIds[i], _principals[i], msg.sender)) {
```

Comment:
For consistency with the single vote version above, shouldn't this revert if the condition is not met?

Reply:
That's a good point, although it would be pretty wasteful to revert a batch of votes if just one cannot be processed (the principal votes at the same time). I would at least add an event when a vote cannot be processed though so we have some visibility

Reply:
I was thinking of returning a list here, or maybe two, with the successful and unsuccessful votes, what do you think?

Reply:
But I guess we need the event for the frontend.

Reply:
Sure, I was thinking of doing both